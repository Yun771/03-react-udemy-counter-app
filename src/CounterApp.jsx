
import { useState } from "react";
import PropTypes from "prop-types";



export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );

    const increment = (e)  => {        // console.log(e);

        setCounter(  counter + 1 );
    };

    const decrement = (e)  => {        // console.log(e);

        setCounter(  counter - 1 );
    };

    const reset = (e) => {
        setCounter(  value );
    };
    


    return <>
        <h1>CounterApp</h1>
        <h2  data-testid="counter">{ counter }</h2>
        
        <button onClick={ increment }>
            +1
        </button>
        <button onClick={ decrement }>
            -1
        </button>
        <button aria-label="btn-reset" onClick={ reset }>
            Reset
        </button>
    </>
    };

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};