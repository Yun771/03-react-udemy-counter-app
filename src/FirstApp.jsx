import PropTypes  from "prop-types";

export const FirstApp = ( { title, subTitle } ) => {

  return (
    <>
      <h1 data-testid='test-title'>{ title }</h1>
      {/* <h1>{newMessage}</h1> */}
      <p>{ subTitle }</p>
      <p>{ subTitle }</p>
    </>
  );
};


FirstApp.propTypes = {
  title: PropTypes.string.isRequired, 
  subTitle: PropTypes.string, 
}

FirstApp.defaultProps = {
  // title: 'No hay titulo', 
  subTitle: 'Subtitulo', 
}