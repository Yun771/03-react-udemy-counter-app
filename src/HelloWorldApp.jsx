// export function App() {
//   return <h1>Hola Mundo</h1>;
// }

import React from "react";

export const HelloWorldApp = () => {
  return <h1>Hello Worl </h1>;
};

// export const HelloWorldApp = () => <h1>Hello World </h1>;
