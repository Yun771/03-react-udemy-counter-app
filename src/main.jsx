import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./CounterApp";
import { FirstApp } from "./FirstApp";
// import { HelloWorldApp } from "./HelloWorldApp";

import './styles.css';

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <FirstApp title="Hola soy Geralt"/> */}
    <CounterApp value={ 0 } />
  </React.StrictMode>
);
