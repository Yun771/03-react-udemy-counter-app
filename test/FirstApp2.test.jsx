import { render, screen } from "@testing-library/react"

import { FirstApp } from '../src/FirstApp'


describe('Pruebas en <FirstApp />', () => { 
    const title = 'Hola, soy Geralt';
    const subtitle = 'Soy un subtitulo';

    test('debe de hacer math con el snapshot', () => { 
        const { container } = render( <FirstApp title={ title } /> )

        expect( container ).toMatchSnapshot();

     });

     test('debe de mostrar el mensaje "Hola, soy Geralt"', () => { 

         
         render( <FirstApp title={ title } /> );
        //  screen.debug();
        
        expect( screen.getByText( title ) ).toBeTruthy();

      });


      test('debe de mostrar el titulo en un h1', () => { 

        render( <FirstApp title={ title } />);

        expect( screen.getByRole("heading",  { level: 1 } ).innerHTML ).toContain( title );



       })

       test('debe de mostrar el subtitulo enviado por props', () => { 
        render(
            <FirstApp
                title={ title }
                subTitle={ subtitle }
            />
        );

        expect( screen.getAllByText(subtitle).length ).toBe( 2 );

       })
})