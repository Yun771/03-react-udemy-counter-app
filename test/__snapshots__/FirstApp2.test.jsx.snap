// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirstApp /> debe de hacer math con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Geralt
  </h1>
  <p>
    Subtitulo
  </p>
  <p>
    Subtitulo
  </p>
</div>
`;
